packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -970,7 +1020,16 @@ export class StripeHelper {
       );
     }
 
-    return validPromotionCodes.includes(code);
+    const valid = await this.validateCouponDurationForPlan(
```

Comment:
this will likely seem like a pedantic (right adjective?) comment but i think it might go a long way in terms of readability/understandability in the future when we have to revisit any of this code. so this method (`checkPromotionCodeForPlan`) is called from both methods to retreive a promo code - the one that checks the valid promo code for plan which is used by previewInvoice and the method that returns the coupon info regardless of validity so that we can determine whether a coupon is expired or redeemed or whatever.

should we consider changing the name of this variable? up to you, the name of the promo codes is already `validPromoCodes` but im wondering if in the future we might wonder why a method that retrieves invalid coupon calls into this and has a variabe called valid which could be true even for a coupon that is invalid in the stripe sense of validity.

Reply:
Thank you for your comments, and I agree a better name would make it more readable.

It also got me to thinking that this is actually the wrong place for this check to happen, since it actually validates the Coupon (coupon duration in this case), and doesn't actually have anything to do with the Promotion Code and shouldn't inhibit retrieving the Promotion Code for a plan.

I was thinking a better place could be the `verifyPromotionAndCoupon` method. What do you think?

Reply:
Good catch, I agree with the above, and took a look at the other commit for the alternative approach. I am not sure when any of these methods will be used in any other context but you're right about the appropriateness of where things are called and it should at least help us with debugging/call stacks later on. 

Feel free to pull those changes into here and just ping me if I need to re-approve. I have already taken a look though, so if my approval stays after the change, feel free to merge when you're ready.